File: python/src/problems1_100/reverse_integer.py
from abc import ABC, abstractmethod
from typing import override


class Problem(ABC):
    @abstractmethod
    def reverse(self, x: int) -> int:
        pass


class Default(Problem):
    @override
    def reverse(self, x: int) -> int:
        negate: bool = False
        s: str = str(x)
        if s[0] == '-':
            s = s[1:]
            negate = True

        if negate:
            return -int(s[::-1])
        return int(s[::-1])


SOLUTIONS: list[Problem] = [Default()]

File: typescript/src/problems1_100/reverse-integer.ts
class Problem {
  reverse(x: number): number {
    throw new Error(`Not implemented (${x}).`);
  }
}

class Default extends Problem {
  reverse(x: number): number {
    let negate = false;
    let s = x.toString();
    if (s[0] === '-') {
      s = s.slice(1);
      negate = true;
    }

    if (negate) {
      return -parseInt(s.split('').reverse().join(''));
    }
    return parseInt(s.split('').reverse().join(''));
  }
}

const SOLUTIONS = [new Default()];
export default SOLUTIONS;
